tests/unit/test_quantize.py

```diff
@@ -225,3 +226,105 @@ def test_quantize_with_empty_and_non_empty_files(tmpdir):
             )
             output_parquet_index_faiss = faiss.read_index(index_parquet_path)
             assert output_parquet_index_faiss.ntotal == len(expected_df)
+
+
+def _search_from_multiple_indices(index_paths, query, k):
+    n = 0
+    all_distances, all_ids, NB_QUERIES = [], [], 1
+    for rest_index_file in index_paths:
+        index = faiss.read_index(rest_index_file)
+        distances, ids = index.search(query, k=k)
+        all_distances.append(distances)
+        all_ids.append(ids + n)
+        n += index.ntotal
+
+    dists_arr = np.stack(all_distances, axis=1).reshape(NB_QUERIES, -1)
+    knn_ids_arr = np.stack(all_ids, axis=1).reshape(NB_QUERIES, -1)
+
+    sorted_k_indices = np.argsort(-dists_arr)[:, :k]
+    sorted_k_dists = np.take_along_axis(dists_arr, sorted_k_indices, axis=1)
+    sorted_k_ids = np.take_along_axis(knn_ids_arr, sorted_k_indices, axis=1)
+    return sorted_k_dists, sorted_k_ids
+
+
+def _merge_indices(index_paths):
+    merged = faiss.read_index(index_paths[0])
+    for rest_index_file in index_paths[1:]:
+        index = faiss.read_index(rest_index_file)
+        faiss.merge_into(merged, index, shift_ids=True)
+    return merged
+
+
+def test_index_correctness_in_distributed_mode_with_multiple_indices(tmpdir):
```

Comment:
This test is unstable due to fact that we use different number of training vectors.
Maybe we can resolve this problem when we have better esitmation of training memory.

Right now current_memory_available is at least 500 MB, it requires us to use larger number of vectors to produce n indices, with the correct estimation of training memory, we could reduce 500MB to the precise value so that we can use less number of vectors and all the vectors are used for training on both n-indcies and groud truth.

Reply:
Fixed on bug on [here](https://github.com/criteo/autofaiss/pull/71/commits/f7be3c711a0081f7e7756ac906ca9e0482068cf3#diff-78ce6ccea7c88ff8a9e0fe85aa7700f75c54c8fda0d325d5c2e32283f2b11a53R496). 
But the tests failed sometimes.

Do not merge the code even if tests get passed !